src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -91,6 +91,45 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException(SR.Arg_BasePathNotFullyQualified);
+
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
+            {
+                return RemoveRelativeSegments(CombineNoChecks(basePath, path), PathInternal.GetRootLength(basePath));
+            }
+            else if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])))
+            {
+                return RemoveRelativeSegments(CombineNoChecks(GetPathRoot(basePath), path.Substring(1)));
+            }
+            else if (length >= 2 && PathInternal.IsValidDriveChar(path[0]) && path[1] == PathInternal.VolumeSeparatorChar)
+            {
+                if (length == 2 || !PathInternal.IsDirectorySeparator(path[2]))
+                {
+                    if (GetPathRoot(path) == GetPathRoot(basePath))
+                    {
+                        return RemoveRelativeSegments(CombineNoChecks(basePath, path.Substring(2)), basePath.Length);
```

Comment:
>CombineNoChecks [](start = 54, length = 15)

We should get rid of the extra allocations in our RemoveRelativeSegments impl (in all places). Use Span to prepare all of the inputs.

Reply:
Sorry, I meant CombineNoChecks() should take Span.

---
In reply to: [162213527](https://github.com/dotnet/coreclr/pull/15579#discussion_r162213527) [](ancestors = 162213527)

Reply:
CombineNoChecks dont have a span implementation

Reply:
It does in [CoreFX](https://github.com/dotnet/corefx/blob/f965141b767517d465c03b899dc87b2e52d2e6fc/src/System.IO.FileSystem/src/System/IO/PathHelpers.cs#L67). Need to port it and use it. :)

Reply:
should I remove the internal string overload for combineNoChecks ?

Reply:
The existing ones in Path can be replaced where the implementation is already done. You can create another issue to deal with the others (such as the array version). (And feel free to take the issue if you want.)